lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/impl/DefaultRemoteManagerImplementation.java

```diff
@@ -107,34 +109,51 @@
       @Parameter(RemoteConfiguration.RetryTimeout.class) final int retryTimeout,
       final LocalAddressProvider localAddressProvider,
       final TransportFactory tpFactory) {
+      this(name, hostAddress, listeningPort, codec, errorHandler, orderingGuarantee, numberOfTries, retryTimeout,
+              localAddressProvider, tpFactory, RangeTcpPortProvider.Default);
+  }
 
-    this.name = name;
-    this.handlerContainer = new HandlerContainer<>(name, codec);
-
-    this.reRecvStage = orderingGuarantee ?
-        new OrderedRemoteReceiverStage(this.handlerContainer, errorHandler) :
-        new RemoteReceiverStage(this.handlerContainer, errorHandler, 10);
-
-    this.transport = tpFactory.newInstance(
-        hostAddress, listeningPort, this.reRecvStage, this.reRecvStage, numberOfTries, retryTimeout);
-
-    this.handlerContainer.setTransport(this.transport);
-
-    this.myIdentifier = new SocketRemoteIdentifier(
-        (InetSocketAddress) this.transport.getLocalAddress());
-
-    this.reSendStage = new RemoteSenderStage(codec, this.transport, 10);
+    @Inject
+    public <T> DefaultRemoteManagerImplementation(
```

Comment:
This constructor is added to be used by Tang only, right? If so, can we make it `private`?